### PR TITLE
Example: card class example with enums

### PR DESCRIPTION
**Pull request by ghost**
## Description
This is an example of what I had in mind, sometimes it's easier to just share code rather than explain it overly with words :)

Enums are a great way to control input to class constructors and even individual functions. The application wouldn't compile, if for example, I tried to create a card with a made up suit or value. This pattern can save us so much time and effort when trying to validate input. They work especially well with strings.

This was the cleanest approach to a card class that I could think of, it's now reusable for any card game where picture cards are valued 10. 

Feel free to take this as it is, or alter if to match your code preferences. I've purposefully left all the errors for you :)

Next steps:
- Do the minimal to fix the other now broken classes, but don't worry too much about the game being fully operational. We'll work our way up from Card. It just needs to compile without errors for now.
- Create a class called PlayingCardUnitTests or something similar and add a unit test that loops every possible card and verifies it's name and value. Try to think of a clever way to do this without writing 50ish tests.
- Open a pull request for your changes and don't merge until I've had a chance to review it :)

### CODE REVIEW

File: src/main/java/PlayingCard/Card.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+public class Card {
+
+    private final CardValue value;
+    private final CardSuit suit;
+
+    public Card(CardSuit suit, CardValue value) {
+        this.suit = suit;
+        this.value = value;
+    }
+
```
Comment (ghost) [inline src/main/java/PlayingCard/Card.java:12]: Unless it was intended you have unnecessary extra lines here

File: src/test/java/PlayingCardUnitTests.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            if (inDeck.contains("" + card.getSuit() + card.getValue())) {
+                Assertions.fail("Multiple of the same card found in deck");
+            } else {
+                inDeck.add("" + card.getSuit() + card.getValue());
+            }
+        }
+    }
+
+    @Test
+    @DisplayName("Check all correct cards are in deck")
```
Comment (ghost) [inline src/test/java/PlayingCardUnitTests.java]: I see what you were thinking here. Asserting the negative outcome is however an anti-pattern. This test is successful when nothing is asserted. I was thinking of something more like this:
```
    @Test
    @DisplayName("A card will return a name when getName is called on a card constructed with a given a suit and value")
    public void CardWillReturnNameWhenConstructedWithSuitAndValue() {
        for (CardSuit suit : CardSuit.values()) {
            for (CardValue value : CardValue.values()) {
                var card = new Card(suit, value);
                var expectedName = "The " + value.name() + " of " + suit.name();

                Assertions.assertEquals(expectedName, card.getName());
            }
        }
    }

File: src/test/java/PlayingCardUnitTests.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PlayingCardUnitTests {
+
+    @Test
+    @DisplayName("Check if multiple of the same card is found in deck")
+    public void multipleInDeckCheck() {
+        var testDeck = Main.createDeck();
```
Comment (ghost) [inline src/test/java/PlayingCardUnitTests.java]: Strictly speaking you're testing Main here, whereas we're only interested in Card for now

File: src/test/java/UnitTests.java [hunk 4]
```diff
@@ -1,77 +1,78 @@
...
-
-        Assertions.assertEquals(Suit.Diamonds, testDeck.get(0).getSuitName());
-        Assertions.assertEquals(Suit.Hearts, testDeck.get(1).getSuitName());
-        Assertions.assertEquals(Suit.Spades, testDeck.get(2).getSuitName());
-        Assertions.assertEquals(Suit.Clubs, testDeck.get(3).getSuitName());
-        Assertions.assertEquals(Suit.Error, testDeck.get(4).getSuitName());
-        Assertions.assertEquals(Suit.Error, testDeck.get(5).getSuitName());
-    }
-}
+//import PlayingCard.Card;
```
Comment (ghost) [inline src/test/java/UnitTests.java]: Checking in commented code is often a code smell and we should avoid it where possible. Git keeps history of every commit. Providing the code exists in one commit it's safe to delete it. You can always revisit historical commits should you need it again

File: target/classes/Card.class
Comment (ghost) [inline target/classes/Card.class:1]: I'm not sure what target files are but I suspect they're related to a build output. Add these to your .gitignore - we only want to commit readable source code

Comment (PolarDyth) [reply target/classes/Card.class:1]: I have added a lot of the unnecessary stuff to .gitignore but I probably did it incorrectly which is why its still there. Is it `.target/` or `target/`?

File: src/test/java/UnitTests.java (hunk 4 above)
Comment (PolarDyth) [reply src/test/java/UnitTests.java]: That makes sense yeah

File: target/classes/Card.class
Comment (ghost) [reply target/classes/Card.class:1]: `target/*`

File: src/test/java/PlayingCardUnitTests.java (hunk 2 above)
Comment (PolarDyth) [reply src/test/java/PlayingCardUnitTests.java]: I thought my logic on that one was a bit weird but thank you. I can see how I was testing main there with the creating the decks and not exclusively testing Card only which makes sense now

Comment (PolarDyth) [reply src/test/java/PlayingCardUnitTests.java]: Would you say there are any more tests that I should look to add to Card?

Comment (ghost) [reply src/test/java/PlayingCardUnitTests.java]: You could add two more, one for getValue and one for getSuit that simply create a card and assert the functions return what is expected. But it's not necessary as getName uses both. Up to you.

**Pull request merged**